### ISSUE TITLE
Add an self-check page for site admin

### ISSUE THREAD

**Issue by wxiaoguang** [go-gitea/gitea#27011]
1. Show deprecated option warnings
1. Check database: version, charset
1. Check whehter AppURL/ROOT_URL is right
1. Check reverse proxy rewriting and sub path (eg: is `%2F` kept?)
1. Check whether Cookie Secure is right
1. Check whether some headers like `X-Forwarded-Porto` are correctly set
1. Check if some directories (eg: work path) are right
1. Check if the server/client time is right (if affects OTP)
1. Check whether there are some legacy files (eg: `/custom/css`, `/custom/robots.txt`, etc)
1. Check whether there are incorrect config options (need to refactor the INI package first)
1. Check whether the queue gets stuck/corrupted
    * #33882
    * #33862
1. Manually version update check

**Comment by lunny** [comment]
All gitea doctor could check?

**Comment by wxiaoguang** [comment]
I guess I have explained this idea not only one time.

And how do you:

```
Check whehter AppURL/ROOT_URL is right
Check reverse proxy rewriting and sub path (eg: is %2F kept?)
Check whether Cookie Secure is right
Check whether some headers like X-Forwarded-Porto are correctly set
Check if the server/client time is right (if affects OTP)
```

??

**Comment by silverwind** [comment]
Maybe `gitea doctor` could output some parseable JSON format for the UI to consume? So these problems could be shown both on UI and when running it on CLI? I think UI is more important as only few people even know about `gitae doctor`.

**Comment by wxiaoguang** [comment]
>> Maybe `gitea doctor` could output some parseable JSON format for the UI to consume?

See the answer above:

> It needs more refactoring work for the doctor system, because some checks consume a lot of time, which are not suitable to be operated by the web UI.

The block problem is not "output", but the whole design, especially the "time". If "check" can't finish in 5-10 seconds, then I think it's not suitable to be put in the self-check page directly, it might need extra work to make it work asynchronously with the web UI.

**Comment by silverwind** [comment]
Yeah, it would need to run the check once on app startup, and then cache the result in-memory.

**Comment by aceslick911** [comment]
fyi might need something that can clear the disk queue folder too as doctor didnt help with my related issue - see https://github.com/go-gitea/gitea/issues/33882#issuecomment-3978803672

> This didnt fix it for me unfortunately - I had a slightly different route to this mess 😓 Basically:
> 
> -> Disk-full event corrupted Gitea's LevelDB queue files. 
> -> All workflows showed first job succeeding, all dependent jobs with a `needs:` ended up permanently stuck in Blocked.
> 

> What fixed it:
> `rm -rf /data/gitea/queues/` (ran on Gitea server pod, then restarted) — forces Gitea to recreate clean queues on startup. Workflows immediately started transitioning correctly.
> 
>  Prevention?
> > i'm switching job_emitter to in-memory queue (no disk, can't corrupt):
> 
> `app.ini`
> 
> ```
>   [queue.job_emitter]
>   TYPE = channel
> ```
